.github/workflows/ci-docs.yaml

```diff
@@ -127,3 +128,66 @@ jobs:
           commit_message: Deploy sol API reference ${{ github.event.head_commit.message }}
           user_name: github-actions[bot]
           user_email: 41898282+github-actions[bot]@users.noreply.github.com
+
+  py-docs:
+    name: py-docs
+    runs-on: ubuntu-latest
+    services:
+      sapphire-localnet-ci:
+        image: ghcr.io/oasisprotocol/sapphire-localnet:latest
+        ports:
+          - 8545:8545
+        options: >-
+          --rm
+          --health-cmd="test -f /CONTAINER_READY"
+          --health-start-period=90s
+    defaults:
+      run:
+        working-directory: ./clients/py
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.12'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install -r requirements.dev.txt
+          pip install sphinx myst-parser
+          make
+          pip install dist/sapphire.py-*.whl
+
+      - name: Build docs
+        run: |
+          mkdir docs
```

Comment:
Add `make docs` rule to the Makefile which builds the api docs and call it here instead.